api/controllers/site.js

```diff
@@ -73,6 +89,12 @@ module.exports = {
       })
       .then(() => S3SiteRemover.removeSite(site))
       .then(() => S3SiteRemover.removeInfrastructure(site))
+      .then(() => {
+        if (Features.enabled(Features.Flags.FEATURE_PROXY_EDGE_DYNAMO)) {
+          return ProxyDataSync.removeSite(site)
```

Comment:
This doesn't necessarily need to be resolved right now.

There is a slight difference in behavior between this implementation, where the promise from `ProxyDataSync.removeSite(site)` is returned as part of the chain, and below when using `async`/`await`.

Here, the next step will not run until `ProxyDataSync.removeSite(site)` completes, but the `async`/`await` version, treats it as "fire and forget".

Perhaps we should make sure these behave consistently.